File: src/main/scala/operator/UnionDemo.scala
package operator

import org.apache.flink.streaming.api.scala._
import org.apache.flink.streaming.api.windowing.time.Time
import source.custorm.NoParalleSource

/**
 * @author: BYDylan
 * @date: 2021/4/22
 * @description: 合并,不限制个数,但限制类型
 */
object UnionDemo {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    val text1: DataStream[Long] = env.addSource(new NoParalleSource)
    val text2: DataStream[Long] = env.addSource(new NoParalleSource)
    val unionall = text1.union(text2)
    val sum: DataStream[Long] = unionall.map(line => {
      println("接收到的数据：" + line)
      line
    }).timeWindowAll(Time.seconds(2)).sum(0)
    sum.print().setParallelism(1)
    env.execute("StreamingDemoUnionScala")
  }
}

File: src/main/java/com/example/transformation/combine/UnionDemo.java
package com.example.transformation.combine;

import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/***
 * @author: BYDylan
 * @date: 2024-09-21 10:30:00
 * @description: 合流: 流的数据类型必须一致, 一次可以合并多条流
 */
public class UnionDemo {
    public static void main(String[] args) throws Exception {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setParallelism(1);

        DataStreamSource<Integer> source1 = env.fromElements(1, 2, 3);
        DataStreamSource<Integer> source2 = env.fromElements(11, 22, 33);
        DataStreamSource<String> source3 = env.fromElements("111", "222", "333");

        DataStream<Integer> union = source1.union(source2).union(source3.map(Integer::valueOf));
        union.print();
        env.execute();
    }
}
